movieapp/src/containers/AddEditMovieForm/AddEditMovieForm.tsx

```diff
@@ -4,64 +4,103 @@ import { Localization } from '../../services/constants';
 import { MovieFormProps } from './AddEditMovieFormTypes';
 import { Movie } from '../../models/Movie';
 import { ButtonTypes } from '../../models/enums';
-import { FormEvent } from 'react';
+import { withFormik, FormikProps, FormikErrors, Form, Field } from 'formik';
 
-export function AddEditMovieForm(props: MovieFormProps): JSX.Element {
-  const { t } = useTranslation();
+interface FormValues extends MovieFormProps {
+  id?: string;
+  title: string;
+  description: string;
+  fullDescription: string;
+}
 
-  const submitForm = (e: FormEvent<HTMLFormElement>): void => {
-    e.preventDefault();
-    const target = e.target as typeof e.target & {
-      id: { value: string };
-      title: { value: string };
-      description: { value: string };
-      fullDescription: { value: string };
-    };
-    const newTitle = target.title.value;
-    const newDescription = target.description.value;
-    const newFullDescription = target.fullDescription.value;
-    const newId = target.id.value || `${newTitle}_${newDescription}_${Math.random().toString()}`;
+const submitForm = (values: FormValues): void => {
+  const newTitle = values.title;
+  const newDescription = values.description;
+  const newFullDescription = values.fullDescription;
+  const newId = values.id || `${newTitle}_${newDescription}_${Math.random().toString()}`;
+
+  const { id, title, description, fullDescription } = new Movie({
+    id: newId,
+    title: newTitle,
+    description: newDescription,
+    fullDescription: newFullDescription
+  });
 
-    const { id, title, description, fullDescription } = new Movie({
-      id: newId,
-      title: newTitle,
-      description: newDescription,
-      fullDescription: newFullDescription
+  if (!values.id) {
+    values.onAddMovie({
```

Comment:
@DrDiman revisit this logic. what is `values` and why does it have these methods

Reply:
I see - HOC provides rest of the props from wrapped component